### PR TITLE
GH Actions: additional workflow for building the PDNS base image using Debian Bookworm.

### PR DESCRIPTION
**Pull request by romeroalx**
Additional workflow for building the PDNS base image using Debian Bookworm.

Usage of separated workflow files to turn older versions on/off when needed.

Tested against the PDNS repository in this [run](https://github.com/romeroalx/pdns/actions/runs/6888921695)

Fixed a wrong comment in the README file about images being private.

Jobs will run now on workers with `ubuntu-22.04`.

### CODE REVIEW

**Review by Habbie — APPROVED** [review]
Looks good :)

I wonder if the yaml files could have less duplication - like you reused the Dockerfile. But we can also do that after merging this.

**Comment by romeroalx** [comment]
Thank you for the review, @Habbie 

The reasoning behind having two different workflows was to be able to turn off the workflow for Debian 11 some time after merging the changes for using the Debian 12 image on the PDNS repo. Still, I am not too attached to that idea.

I will update this pull request and merge both workflow files.

File: .github/workflows/build-debian-images.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  IMAGE_TAG: master
+
+jobs:
+  build-and-push-debian-images:
+    strategy:
+      matrix:
+        include:
+        - image-id: debian-11-pdns-base
+          debian-release-name: bullseye-slim
+        - image-id: debian-bookworm-pdns-base
```
Comment (Habbie) [inline .github/workflows/build-debian-images.yaml]:
```suggestion
        - image-id: debian-12-pdns-base
```

perhaps?

**Review by Habbie — COMMENTED** [review]
one suggestion/question, otherwise good!

File: .github/workflows/build-debian-images.yaml (hunk 1 above)
Comment (romeroalx) [reply .github/workflows/build-debian-images.yaml:19]: That part is a little bit odd. I was trying to remove the references to a version number and move to a version name without being too disruptive (our PDNS repo currently points to `debian-11-pdns-base`).

Maybe I should update all references to a version number like this:

```
        - image-id: debian-11-pdns-base
          debian-release-name: 11-slim
        - image-id: debian-12-pdns-base
          debian-release-name: 12-slim
```

Any preference?

Comment (romeroalx) [reply .github/workflows/build-debian-images.yaml]: Ok, done!

**Pull request merged**